[IMP] account_financial_report: let store auto-generated legal sequence number

yajo opened the pull request:
- Let accountants choose to store auto-generated sequence.
- Let them use that sequence number in search, form and tree views of account.move.
- Fix one bug that was forbidding rendering report.

With this improvement, accountants will be able to find data when referenced by the official legal number.

@moduon MT-672

JordiBForgeFlow commented:
@yajo thanks for the proposal. I suggest to add a module "account_financial_report_auto_sequence". This feature is not used anywhere but in Spain, as far as I can see, and many countries use the financial reports nowadays. They will be very confused with this feature.

pedrobaeza commented:
I don't think this is needed at all, as it's for a very specific requirement. You can have the PDF sent to the "Registro Mercantil" in case there's any audit to match one number with the other, but most of the checks can be done with the real number and ignore the other. Take into account that the progressive and consecutive numbering is a requirement from the old 1887 merchant code.

yajo commented:
OK, I'll add it in a separate module.

rafaelbn commented:
@JordiBForgeFlow , do you mean this PR should be also taken out from this module? Thank you

[[13.0][IMP] account_financial_report: added 'auto sequence' option in Journal Ledger. #787](https://github.com/OCA/account-financial-reporting/pull/787)

[[14.0][IMP] account_financial_report: added 'auto sequence' option in Journal Ledger](https://github.com/OCA/account-financial-reporting/pull/798) #798

JordiBForgeFlow commented:
@yajo I did not notice those PR's... I would have preferred to have those also exist in a separate module, but they are now merged.

I can see that in your case the split is rather simple

yajo committed:
[ADD] account_financial_report_remember_sequence: let store auto-generated legal sequence number

- Let accountants choose to store auto-generated sequence.
- Let them use that sequence number in search, form and tree views of account.move.
- Fix one bug that was forbidding rendering report.

With this addon, accountants will be able to find data when referenced by the official legal number in some countries.

@moduon MT-672

pedrobaeza commented:
Please don't ignore my comments. IMO everything should stay as is. No change is needed.

Shide commented:
I think this change makes Accountants job easy when searching for specific account moves on an audit.
The audit person may only have the "legal" number (000123) and not the Odoo one. And avoids to store the Excel file or the PDF file forever. You are sure that the presented "legal" number is stored on the database.

yajo commented:
I have made the split, now all is in a separate module (except the bugfix). There's still some work to do so I'll set it as draft for now.

rafaelbn commented:
Hello! 😄 

We don't ignore anyone, in fact we have split it as @yajo  said in https://github.com/OCA/account-financial-reporting/pull/876#issuecomment-1113231274

- a FIX of the module
- a new module `account_financial_report_remember_sequence` (name was temporally and that is why **this PR is Draft**)

In fact, I agree with @JordiBForgeFlow https://github.com/OCA/account-financial-reporting/pull/876#issuecomment-1113150421

About if is needed or not, there are many modules in OCA with the need of different clients and any one can choose the module the use.

Said that, and as Jordi said. Including [these PR](https://github.com/OCA/account-financial-reporting/pull/876#issuecomment-1113154733) on a global module is done and we cannot change that, but an externa module `account_financial_report_auto_sequence` would be better (only) in my opinion.

Thank you for all you opinions 🙏 👍 . We sleep it over (and ask again the auditors are requesting it) to develop the best option.

Best regards

pedrobaeza commented:
Ok if you want to go ahead with the storage not breaking existing things. I tried to explain why I think it's not needed to do it, but of course you decide.

ValentinVinagre commented on account_financial_report/report/templates/journal_ledger.xml:
```diff
@@ -306,10 +306,7 @@
-                    t-if="move_line['amount_currency']"
-                    t-options="{'widget': 'float', 'precision': 2}"
-                >
+                <t t-if="move_line['amount_currency']">
```
Why did you delete `t-option`?

yajo replied:
It was giving a qweb exception while printing the report. It didn't have any sense, because it's just used for formatting, but no output is ever made by a `<t t-if` node, so it was an obvious bug.

ValentinVinagre commented on account_financial_report_remember_sequence/report/journal_ledger.py:
```diff
@@ -0,0 +1,18 @@
+        )
+        if wizard.with_auto_sequence and wizard.store_auto_sequence:
+            move = self.env["account.move"].browse(result["move_id"])
+            move.legal_sequence = auto_sequence
```
If an "account.move" has N "account.move.line", aren't you overwriting "legal_sequence" field N times?

yajo replied:
Yes, but since v13 it's a cheap operation.

ValentinVinagre reviewed (commented):
Some Comments.

HaraldPanten reviewed (commented):
Do you want to keep it as a sequence? Don't you prefer to keep the number as it is? I mean... "00003" instead of just "3"?


![Captura de Pantalla 2022-04-29 a las 17 02 30](https://user-images.githubusercontent.com/32449357/165971575-c5df669a-6251-429f-993e-093592ee104d.png)

yajo commented:
We proposed this as a MVP to fix the problem expressed in https://github.com/OCA/account-financial-reporting/pull/876#issuecomment-1113199549. However we examined the issue with more depth and found out that the MVP is just not enough for our use case, so I'll close this PR while we get to a better solution.

We initially thought this was a global need due to https://github.com/OCA/account-financial-reporting/pull/787 being included in the base module, but after what's been talked here, we'll do a separate module to avoid polluting the main one, so whoever don't need it won't have a problem.

Sorry for the noise and thanks everyone for your reviews!
